app/javascript/packs/modPage.js

```diff
@@ -0,0 +1,100 @@
+const buttons = document.querySelectorAll('.reaction-button, .reaction-vomit-button');
+
+function applyReactedClass(category) {
+  const upVote = document.querySelector("[data-category='thumbsup']");
+  const downVote = document.querySelector("[data-category='thumbsdown']");
+  const vomitVote = document.querySelector("[data-category='vomit']");
+
+  if (category === 'thumbsup') {
+    downVote.classList.remove('reacted');
+    vomitVote.classList.remove('reacted');
+  } else {
+    upVote.classList.remove('reacted');
+  }
+}
+
+for (let i = 0; i < buttons.length; i++) {
+  let button = buttons[i];
+  button.onclick = function (event) {
+    event.preventDefault();
+    let thisButton = this;
+    applyReactedClass(thisButton.dataset.category);
+    thisButton.classList.add('reacted');
+
+    function successCb(response) {
+      if (response.result === 'create') {
+        thisButton.classList.add('reacted');
+      } else {
+        thisButton.classList.remove('reacted');
+      }
+    }
+
+    let formData = new FormData();
+    formData.append('reactable_type', thisButton.dataset.reactableType);
+    formData.append('category', thisButton.dataset.category);
+    formData.append('reactable_id', thisButton.dataset.reactableId);
+
+    getCsrfToken()
+      .then(sendFetch('reaction-creation', formData))
+      .then(function (response) {
+        if (response.status === 200) {
+          response.json().then(successCb);
+        }
+      });
+  };
```

Comment:
I feel like you can refactor this :) maybe in a timebox of an hour? If it takes too long we don't need to ship with the refactored version.

Reply:
Yup I shall do so in separate PR 😄